shared/reducers/fs.js

```diff
@@ -27,6 +27,13 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
     case FsGen.filePreviewLoaded:
       return state.updateIn(['pathItems', action.payload.path], (original: Types.PathItem) => {
         const {meta} = action.payload
+
+        // Since updateIn passes `original` in as `any`, it may not be an
+        // actual PathItem.
+        if (!original) {
+          return meta
```

Comment:
Will this work to set it even if it doesn't exist?

Reply:
Yup:

```
> (new I.Map({})).updateIn(['a'], old=>1)
Map { "a": 1 }
```